client/base_client.go

```diff
@@ -55,10 +55,10 @@ type baseClient struct {
 
 	security SecurityOption
 
-	gRPCDialOptions  []grpc.DialOption
-	timeout          time.Duration
-	maxRetryTimes    int
-	enableForwarding bool
+	// BaseClient options.
+	gRPCDialOptions []grpc.DialOption
+	timeout         time.Duration
+	maxRetryTimes   int
```

Comment:
only move out enableForwarding?

Reply:
Because the usage scope of `enableForwarding` is in `client`, and the others in `baseClient`. We also need to consider the scope of the new configuration in the future before adding it to the corresponding scope. For example, `maxBatchWaitInterval` is an option that should be placed in the `client` rather than `baseClient`.

Reply:
https://github.com/tikv/pd/pull/4054#discussion_r706986632

Reply:
The base client is abstracted as a common part for dynamic configuration. Since that feature is changed a lot, there is no need to distinguish between client and base client.

Reply:
I think this change is more about code style. Putting different options into its scope can help understand the role of an option better.